Userland/Libraries/LibLine/SuggestionManager.cpp

```diff
@@ -10,55 +10,46 @@
 namespace Line {
 
 CompletionSuggestion::CompletionSuggestion(StringView completion, StringView trailing_trivia, StringView display_trivia, Style style)
-    : style(style)
-    , text_string(completion)
-    , display_trivia_string(display_trivia)
+    : text(MUST(String::from_utf8(completion)))
+    , trailing_trivia(MUST(String::from_utf8(trailing_trivia)))
+    , display_trivia(MUST(String::from_utf8(display_trivia)))
+    , style(style)
     , is_valid(true)
 {
-    Utf8View text_u8 { completion };
-    Utf8View trivia_u8 { trailing_trivia };
-    Utf8View display_u8 { display_trivia };
-
-    for (auto cp : text_u8)
-        text.append(cp);
-
-    for (auto cp : trivia_u8)
-        this->trailing_trivia.append(cp);
-
-    for (auto cp : display_u8)
-        this->display_trivia.append(cp);
-
-    text_view = Utf32View { text.data(), text.size() };
-    trivia_view = Utf32View { this->trailing_trivia.data(), this->trailing_trivia.size() };
-    display_trivia_view = Utf32View { this->display_trivia.data(), this->display_trivia.size() };
 }
 
 void SuggestionManager::set_suggestions(Vector<CompletionSuggestion>&& suggestions)
 {
-    m_suggestions = move(suggestions);
+    enum IndexError {
+        NotFoundError
+    };
+    auto code_point_at = [](Utf8View view, size_t index) -> ErrorOr<u32, IndexError> {
+        size_t count = { 0 };
+        for (auto cp : view) {
+            if (count == index) {
+                return cp;
+            }
+            count++;
+        }
+        return IndexError::NotFoundError;
```

Comment:
All users of this use MUST() anyway, just make this return `u32` and `VERIFY_NOT_REACHED()` here.